controller/lib/src/external_api/workload/controller.rs

```diff
@@ -1,46 +1,318 @@
-    service::get_workloads(&["1".to_string(), "2".to_string(), "3".to_string()]);
-    HttpResponse::Ok().body("workload")
-}
+use super::{model::Pagination, model::WorkloadDTO, model::WorkloadError, service};
+use actix_web::http::StatusCode;
+use actix_web::{web, HttpResponse, Responder, Scope};
 
-#[get("/{workload_id}")]
-pub async fn get_workload(workload_id: web::Path<String>) -> impl Responder {
-    service::get_workloads(&[workload_id.to_string()]);
-    HttpResponse::Ok().body(format!("workload {}", workload_id))
-}
+struct WorkloadController {}
 
-#[put("/")]
-pub async fn put_workload(body: web::Json<WorkloadInfo>) -> impl Responder {
-    let workload_info = body.into_inner();
-    service::create_workload(&workload_info);
-    HttpResponse::Ok().body("put_workload")
-}
+impl WorkloadController {
+    pub fn get_services(&self) -> Scope {
+        web::scope("/workload")
+            .service(
+                web::resource("/{namespace}/{workload_id}")
+                    .route(web::delete().to(WorkloadController::delete_workload))
+                    .route(web::get().to(WorkloadController::get_workload))
+                    .route(web::patch().to(WorkloadController::patch_workload)),
+            )
+            .service(
+                web::resource("/{namespace}")
+                    .route(web::put().to(WorkloadController::put_workload))
+                    .route(web::get().to(WorkloadController::get_all_workloads)),
+            )
+    }
+
+    /// `get_workload` is an async function that handle **/workload/\<namespace>/<workload_id>** route (GET)
+    /// # Description:
+    /// * Get a workload by id and namespace
+    /// # Arguments:
+    ///
+    /// * `params`: web::Path<(String, String)> - The first Path parameter is the namespace and the second the workload id.
+    pub async fn get_workload(
+        params: web::Path<(String, String)>,
+        data: web::Data<ActixAppState>,
+    ) -> impl Responder {
+        let (namespace, workload_id) = params.into_inner();
+
+        let mut workload_service = service::WorkloadService::new(&data.etcd_address)
+            .await
+            .map_err(|err| {
+                let mut http_response = HttpResponse::InternalServerError();
+                match err {
+                    WorkloadError::Etcd(msg) => http_response.body(format!("Etcd error: {} ", msg)),
+                    _ => http_response.body("Internal Server Error"),
+                }
+            })
+            .unwrap();
+
+        match workload_service
+            .get_workload(&workload_id, &namespace)
+            .await
+        {
+            Ok(workload) => match serde_json::to_string(&workload) {
+                Ok(json) => HttpResponse::build(StatusCode::OK).body(json),
+                Err(err) => HttpResponse::InternalServerError().body(format!(
+                    "Error while converting the workload to json string : {}",
+                    err
+                )),
+            },
```

Comment:
This patterns repeats several times across this PR, you should create a `Workload` method that converts `self` to a `HttpResponse` (and take a `StatusCode` as an input parameter) and use that method instead of having several copies of that code snippet.

Reply:
I implemented the function to_http for Workload struct.
It serialize the workload in JSON, if failed return a http error, else it return the json in a http response.
What do you think about it ?